packages/path_provider/lib/path_provider.dart

```diff
@@ -106,3 +111,64 @@ Future<Directory> getExternalStorageDirectory() async {
   }
   return Directory(path);
 }
+
+/// Paths to directories where application specific cache data can be stored.
+/// These paths typically reside on external storage like separate partitions
+/// or SD cards. Phones may have multiple storage directories available.
+///
+/// The current operating system should be determined before issuing this
+/// function call, as this functionality is only available on Android.
+///
+/// On iOS, this function throws an UnsupportedError as it is not possible
+/// to access outside the app's sandbox.
+///
+/// On Android this returns Context.getExternalCacheDirs() or
+/// Context.getExternalCacheDir() on API levels below 19.
+Future<List<Directory>> getExternalCacheDirectories() async {
+  if (_platform.isIOS)
+    throw UnsupportedError("Functionality not available on iOS");
+  final List<String> paths =
+      await _channel.invokeListMethod<String>('getExternalCacheDirectories');
+
+  return paths.map((String path) => Directory(path)).toList();
+}
+
+/// Shadows directory values from Androids `android.os.Environment` class.
+///
+/// https://developer.android.com/reference/android/os/Environment.html#fields_1
+class AndroidEnvironment {
```

Comment:
- Users are advised to use a string to determine those specific folders and I am wondering where to store the string.
 - In `android_intent`, we have a list of actions in Dart (using their actual Android intent values) - see https://github.com/flutter/plugins/blob/master/packages/android_intent/lib/flag.dart

It does make sense but developers could use any other string (and may not easily discover the AndroidEnvironment class for the constants).

If we turn it into a enum (and let path provider map those values to String values internally), it would increase the complexity quite a bit.

@amirh do you think the case here is the same as in `android_intent`?

Reply:
I spoke to @amirh  offline and we do prefer to have it as enum in the public api. And we can have a method convert the enum to String in dart before sending it to the Android Platform.